From 16be8d6518a2c25b578078495d5f6bea397e2050 Mon Sep 17 00:00:00 2001
From: Amine Khaldi <amine.khaldi@reactos.org>
Date: Wed, 4 Feb 2015 16:15:51 +0100
Subject: [PATCH 1/7] Mark this as ReactOS modified source.

---
 Source/CMakeVersionCompute.cmake | 1 +
 1 file changed, 1 insertion(+)

diff --git a/Source/CMakeVersionCompute.cmake b/Source/CMakeVersionCompute.cmake
index 496d6cf1a..15d8734bc 100644
--- a/Source/CMakeVersionCompute.cmake
+++ b/Source/CMakeVersionCompute.cmake
@@ -15,6 +15,7 @@ set(CMake_VERSION ${CMake_VERSION_MAJOR}.${CMake_VERSION_MINOR}.${CMake_VERSION_
 if(CMake_VERSION_RC)
   set(CMake_VERSION ${CMake_VERSION}-rc${CMake_VERSION_RC})
 endif()
+set(CMake_VERSION_SOURCE "ReactOS")
 if(CMake_VERSION_SOURCE)
   set(CMake_VERSION ${CMake_VERSION}-${CMake_VERSION_SOURCE})
 endif()
-- 
2.17.1

