From 2cdd02549cdbef42d8257f25cc8d2146b698e809 Mon Sep 17 00:00:00 2001
From: Amine Khaldi <amine.khaldi@reactos.org>
Date: Wed, 4 Feb 2015 16:16:26 +0100
Subject: [PATCH 2/6] Account for XP compatibility when linking the CMake
 binaries.

---
 Source/CMakeLists.txt | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/Source/CMakeLists.txt b/Source/CMakeLists.txt
index 2835ee65a8..b0883f2565 100644
--- a/Source/CMakeLists.txt
+++ b/Source/CMakeLists.txt
@@ -753,6 +753,9 @@ if(WIN32 AND NOT CYGWIN)
   set_source_files_properties(cmcldeps.cxx PROPERTIES COMPILE_DEFINITIONS _WIN32_WINNT=0x0501)
   add_executable(cmcldeps cmcldeps.cxx ${MANIFEST_FILE})
   list(APPEND _tools cmcldeps)
+  if(MSVC)
+    set_property(TARGET cmcldeps APPEND_STRING PROPERTY LINK_FLAGS "/SUBSYSTEM:CONSOLE,5.01")
+  endif()
   target_link_libraries(cmcldeps CMakeLib)
 endif()
 
@@ -987,6 +990,9 @@ endif()
 # Build CMake executable
 add_executable(cmake cmakemain.cxx cmcmd.cxx cmcmd.h ${MANIFEST_FILE})
 list(APPEND _tools cmake)
+if(MSVC)
+  set_property(TARGET cmake APPEND_STRING PROPERTY LINK_FLAGS "/SUBSYSTEM:CONSOLE,5.01")
+endif()
 target_link_libraries(cmake CMakeLib)
 
 if(CMake_ENABLE_SERVER_MODE)
@@ -1005,11 +1011,17 @@ endif()
 # Build CTest executable
 add_executable(ctest ctest.cxx ${MANIFEST_FILE})
 list(APPEND _tools ctest)
+if(MSVC)
+  set_property(TARGET ctest APPEND_STRING PROPERTY LINK_FLAGS "/SUBSYSTEM:CONSOLE,5.01")
+endif()
 target_link_libraries(ctest CTestLib)
 
 # Build CPack executable
 add_executable(cpack CPack/cpack.cxx ${MANIFEST_FILE})
 list(APPEND _tools cpack)
+if(MSVC)
+  set_property(TARGET cpack APPEND_STRING PROPERTY LINK_FLAGS "/SUBSYSTEM:CONSOLE,5.01")
+endif()
 target_link_libraries(cpack CPackLib)
 
 # Curses GUI
-- 
2.20.1

